src/services/api/routes/communities.ts

```diff
@@ -22,27 +22,31 @@ import { ParamPack } from '@/types/common/param-pack';
 
 const createNewCommunitySchema = z.object({
 	name: z.string(),
-	description: z.string(),
+	description: z.string().optional(),
 	icon: z.string(),
-	app_data: z.string()
+	app_data: z.string().optional()
 });
 
 const router: express.Router = express.Router();
 
-function respondCommunityNotFound(response: express.Response) : void {
-	response.status(404);
-	response.send(xmlbuilder.create({
+function respondCommunityError(response: express.Response, http_code: number, error_code: number): void {
```

Comment:
while technically "http code" is accurate, it's a bit awkward as the full name for this value is "http _status_ code", usually shortened to "status code" or just "status"